web/src/app/pages/main-page-container/main-page/map/map.component.ts

```diff
@@ -152,6 +157,10 @@ export class MapComponent implements AfterViewInit, OnDestroy {
     if (this.disableMapClicks) {
       return;
     }
+    if (!this.shouldEnableDrawingTools) {
```

Comment:
I think a slightly cleaner way you could do this is to set editMode to None if drawing tools are not enabled. So you could use a ternary expression below to set to none if drawing tools not enabled or get edit modes otherwise.

Reply:
Sorry, I merged before I saw this. This sounds good, I'll send another PR to address this.